cross/org.eclipse.cdt.build.crossgcc/src/org/eclipse/cdt/internal/build/crossgcc/CrossBuildMacroSupplier.java

```diff
@@ -0,0 +1,91 @@
+/*******************************************************************************
+ * Copyright (c) 2023 John Dallaway and others.
+ *
+ * This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License 2.0
+ * which accompanies this distribution, and is available at
+ * https://www.eclipse.org/legal/epl-2.0/
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ *
+ * Contributors:
+ *     John Dallaway - initial API and implementation (#361)
+ *******************************************************************************/
+package org.eclipse.cdt.internal.build.crossgcc;
+
+import org.eclipse.cdt.managedbuilder.core.BuildException;
+import org.eclipse.cdt.managedbuilder.core.IConfiguration;
+import org.eclipse.cdt.managedbuilder.core.IOption;
+import org.eclipse.cdt.managedbuilder.macros.BuildMacroException;
+import org.eclipse.cdt.managedbuilder.macros.IBuildMacro;
+import org.eclipse.cdt.managedbuilder.macros.IBuildMacroProvider;
+import org.eclipse.cdt.managedbuilder.macros.IConfigurationBuildMacroSupplier;
+import org.eclipse.cdt.utils.IGnuToolFactory;
+import org.eclipse.core.runtime.Status;
+
+public class CrossBuildMacroSupplier implements IConfigurationBuildMacroSupplier {
+
+	/**
+	 * A CDT build variable defining the GNU tool prefix for use by
+	 * an {@link org.eclipse.cdt.utils.IGnuToolFactory} implementation.
+	 */
+	private static class GnuToolPrefixMacro implements IBuildMacro {
```

Comment:
Would it be possible to put this class as a public class (public API) instead with the option id as a parameter to the constructor?
If left as is, then a downstream vendor with a custom toolchain definition would need to more or less copy this implementation and put all that in there code base. That by itself might not be an issue, but if the code base is not EPL-2.0, then it will be a mess for the vendor and perhaps would require them to have 1 additional plugin just to have the proper license on the single class.

Reply:
How about an `AbstractGnuToolPrefixMacro` class which is missing the `getStringValue()` method tucked in to the Java package alongside `IGnuToolFactory`?

Reply:
I've now looked at implementing an abstract class. The benefits are modest but probably worthwhile. Does this work for you?
```
package org.eclipse.cdt.managedbuilder.macros;

import org.eclipse.cdt.utils.IGnuToolFactory;
import org.eclipse.core.runtime.Status;

/**
 * A CDT build variable defining the GNU tool prefix for use by
 * an {@link org.eclipse.cdt.utils.IGnuToolFactory} implementation.
 *
 * @since 9.6
 */
public abstract class AbstractGnuToolPrefixMacro implements IBuildMacro {

	@Override
	public String getName() {
		return IGnuToolFactory.GNU_TOOL_PREFIX_VARIABLE;
	}

	@Override
	public int getValueType() {
		return IBuildMacro.VALUE_TEXT;
	}

	@Override
	public int getMacroValueType() {
		return getValueType();
	}

	@Override
	public String[] getStringListValue() throws BuildMacroException {
		throw new BuildMacroException(
				Status.error("Not a string list build macro: " + IGnuToolFactory.GNU_TOOL_PREFIX_VARIABLE)); //$NON-NLS-1$
	}

	@Override
	abstract public String getStringValue() throws BuildMacroException;
}
```

Reply:
I thought about passing an `IOption` through the constructor, but this seems too limiting. Some implementations may not use an `IOption`. Perhaps they provide a constant value or obtain the value by other means such as an `IPreferenceService`.

Reply:
We could provide a helper method (callable by the concrete implementation) in the abstract class:
```
public String getStringValue(IOption option) throws BuildMacroException {
	try {
		return option.getStringValue();
	} catch (BuildException e) {
		throw new BuildMacroException(Status.error(
				"Error getting build macro string value: " + IGnuToolFactory.GNU_TOOL_PREFIX_VARIABLE, e)); //$NON-NLS-1$
	}
}
```

Reply:
@Torbjorn-Svensson, I've implemented the abstract class with `IOption` helper method and updated the PR accordingly.